src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -133,39 +133,39 @@ namespace OpenMS
   {
   }
 
-
   void TheoreticalSpectrumGenerator::getSpectrum(PeakSpectrum& spectrum, const AASequence& peptide, Int min_charge, Int max_charge) const
   {
     if (peptide.empty())
     {
       return;
     }
 
+    MSSpectrum::Chunks chunks(spectrum);
     PeakSpectrum::StringDataArray ion_names;
     PeakSpectrum::IntegerDataArray charges;
 
     if (add_metainfo_)
     {
       if (spectrum.getIntegerDataArrays().size() > 0)
       {
-        charges = spectrum.getIntegerDataArrays()[0];
+        charges = std::move(spectrum.getIntegerDataArrays()[0]);
       }
       if (spectrum.getStringDataArrays().size() > 0)
       {
-        ion_names = spectrum.getStringDataArrays()[0];
+        ion_names = std::move(spectrum.getStringDataArrays()[0]);
```

Comment:
why is there so much moving? would it be possible to work with pointer or references PeakSpectrum::IntegerDataArray* charges ?

Reply:
I managed to use pointers. There is less moving now.